sdk/core/src/CommonNetworkMember/BaseNetworkMember.ts

```diff
@@ -210,26 +206,14 @@ export abstract class BaseNetworkMember {
     password: string,
     keyOptions?: KeyOptions,
   ) {
+    const {
+      basicOptions: { registryUrl },
+      accessApiKey,
+    } = options
+    const api = new RegistryApiService({ registryUrl, accessApiKey, sdkVersion: extractSDKVersion() })
+    const didResolver = new DidResolver({ registryUrl, accessApiKey, sdkVersion: extractSDKVersion() })
     const didMethod = options.otherOptions.didMethod || DEFAULT_DID_METHOD
-    const passwordBuffer = KeysService.normalizePassword(password)
-    const seedWithMethod = await generateFullSeed(dependencies.platformCryptographyTools, didMethod, keyOptions)
-    const encryptedSeed = await KeysService.encryptSeed(seedWithMethod, passwordBuffer)
-    const keysService = new KeysService(encryptedSeed, password)
-
-    const didDocumentService = DidDocumentService.createDidDocumentService(keysService)
-    const didDocument = await didDocumentService.buildDidDocument(
-      new DidResolver({
-        registryUrl: options.basicOptions.registryUrl,
-        accessApiKey: options.accessApiKey,
-        sdkVersion: extractSDKVersion(),
-      }),
-    )
-    const did = didDocument.id
-    const didDocumentKeyId = didDocumentService.getKeyId()
-
-    await BaseNetworkMember._anchorDid(encryptedSeed, password, didDocument, 0, options)
-
-    return { did, didDocumentKeyId, encryptedSeed }
+    return register(api, didResolver, didMethod, dependencies.platformCryptographyTools, password, keyOptions)
```

Comment:
Are we handling error which thrown by this method or it consumer responsibility?

Reply:
yes, as it was before. The main goal of this PR to add an 'elem-anchored' feature, not make refactoring